sphinx/deprecation.py

```diff
@@ -44,10 +44,13 @@ def __init__(self, module: Any, modname: str, objects: Dict, warning: "Type[Warn
 
     def __getattr__(self, name: str) -> Any:
         if name in self._objects:
-            warnings.warn("%s.%s is deprecated. Check CHANGES for Sphinx "
-                          "API modifications." % (self._modname, name),
-                          self._warning, stacklevel=3)
-            return self._objects[name]
+            obj = self._objects[name]
+            warnings.warn(
+                "The alias '{}.{}' is deprecated, use '{}.{}' instead. Check CHANGES for "
+                "Sphinx API modifications.".format(
+                    self._modname, name, obj.__module__, obj.__name__),
```

Comment:
We use `deprecated_alias` not only for class and functions. Is this working fine for constants and other objects?

Reply:
Okay, I've reworked it so the canonical name is passed to `deprecated_alias`. Going through all the calls also revealed instances where is was being used as a full deprecation so I've separated that use case into `deprecated_attribute`.